src/AutoRest.CSharp/Common/Output/Models/ClientFields.cs

```diff
@@ -77,7 +77,7 @@ public ClientFields(IEnumerable<Parameter> parameters, BuildContext? context = n
 
             foreach (Parameter parameter in parameters)
             {
-                var field = parameter.IsResourceIdentifier
+                var field = parameter.IsResourceIdentifier || parameter.Name == "endpoint"
```

Comment:
Can we not use hard-coded 'endpoint' here? 
And the code is out-of-date, need to rebase.